### PR TITLE
Thunk to make API call

### PR DESCRIPTION
**Pull request by rottencandy**
- Add `redux-thunk` middleware
- Turn `searchName` action into thunk, which fetches details by making API call
- Dispatch `searchName` on submitting form
- Update store with returned data, fetch status

### CODE REVIEW

File: src/redux/actionCreators.ts [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
 
-export function searchName (name: string): SearchNameType {
-  return { type: SEARCH_NAME, payload: { name } }
+export function requestDetails (name: string): RequestDetailsType {
+  return { type: REQUEST_DETAILS, payload: { name } }
+}
+
+type RawVictimDetail = { map: VictimDetail };
+export function recieveDetails (
+  wasSuccessful: boolean,
```
Comment (invincibleJai) [inline src/redux/actionCreators.ts]: nit:  isSuccess

File: src/redux/actionCreators.ts [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+    payload: { wasSuccessful, data: parsedData }
+  }
+}
+
+export function searchName (name: string): AppThunk {
+  return function (dispatch: Dispatch): void {
+    dispatch(requestDetails(name))
+    if (process.env.REACT_APP_MOCK_API) {
+      const mockResponse = getVictimData(name)
+      console.log(mockResponse)
```
Comment (invincibleJai) [inline src/redux/actionCreators.ts]: remove `console`

File: src/redux/reducers/details.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  REQUEST_DETAILS,
+  RECIEVE_DETAILS,
+  RequestDetailsType,
+  RecieveDetailsType
+} from '../types'
+
+const initialState = {
+  isFetching: false,
+  wasSuccessful: true,
+  hasData: false,
```
Comment (invincibleJai) [inline src/redux/reducers/details.ts]: is this needed?

File: src/redux/reducers/details.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  data: []
+}
+
+type ActionType = RequestDetailsType & RecieveDetailsType;
+export function detailsReducer (state = initialState, action: ActionType) {
+  switch (action.type) {
+    case REQUEST_DETAILS:
+      return Object.assign(state, { isFetching: true })
```
Comment (invincibleJai) [inline src/redux/reducers/details.ts]: nit: can use destructuring

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  lon: string;
+  medicalNeeded: boolean;
+  numberOfPeople: number;
+  victimName: string;
+  victimPhoneNumber: string;
+  timeStamp: number;
+  status: string;
+}
+
+export default interface SearchInput {
```
Comment (invincibleJai) [inline src/types.ts]: do we need it to be default export?

File: src/redux/actionCreators.ts (hunk 1 above)
Comment (rottencandy) [reply src/redux/actionCreators.ts]: Changed

File: src/redux/actionCreators.ts (hunk 2 above)
Comment (rottencandy) [reply src/redux/actionCreators.ts]: Removed

File: src/redux/reducers/details.ts (hunk 3 above)
Comment (rottencandy) [reply src/redux/reducers/details.ts]: Removed

File: src/redux/reducers/details.ts (hunk 4 above)
Comment (rottencandy) [reply src/redux/reducers/details.ts]: Updated

File: src/types.ts (hunk 5 above)
Comment (rottencandy) [reply src/types.ts]: This is only here to satisfy the old code. This will not be needed after the refactor code is merged.

**Pull request merged**